statsd/statsd.go

```diff
@@ -489,15 +489,23 @@ func (c *Client) SimpleServiceCheck(name string, status ServiceCheckStatus) erro
 	return c.ServiceCheck(sc)
 }
 
+func (c *Client) shutdown() {
+	c.Lock()
```

Comment:
I don't think we need to hold the client lock here

Reply:
This was here to prevent a race between two routines calling `Close()` at the same time. 

On closer inspection, I realize it's already illegal to call `Close()` more than once, so this is not necessary.

Reply:
If users can do something, it's not illegal. But I suspect @arbll is right and this doesn't need a lock, even if called in parallel. Channels are concurrent safe, what are we guarding against?

As a side note, if we wanna go completely leak-free on the close, we should also fix this bit: https://github.com/DataDog/datadog-go/blob/87e3273d8c230a3abfbbe9dc16b9aa935b3af496/statsd/sender.go#L114

It also closes, but doesn't wait for the goroutines to exit. Here, we can use a simpler mechanism, similar to what we have in our own tracer:
```
s.stop <- struct{}
<-s.stop
```
The first send on the stop channel will cause the `sendLoop` to exit, and the receive will be triggered once the goroutine completely if the channel closes in a defer around [this line](https://github.com/DataDog/datadog-go/blob/87e3273d8c230a3abfbbe9dc16b9aa935b3af496/statsd/sender.go#L90):
```
defer close(s.stop)
```

We use similar mechanisms in dd-trace-go

Reply:
👍 I'll fix the sender panic here as well.

The lock is necessary because multiple goroutines can go through the select's default branch at once.
There's no assurance that a goroutine that takes the default branch will call `close(c.stop)` before another goroutine gets through select's default branch, and so it's possible for more than one goroutine to call `close(c.stop)` unless the select and close are protected.

Reply:
> The lock is necessary because multiple goroutines can go through the select's default branch at once.

Indeed. Very correct 👍